backend/homework/6/src/main/java/org/example/ServiceVehicle.java

```diff
@@ -0,0 +1,38 @@
+package org.example;
+
+import jakarta.annotation.PostConstruct;
+
+import org.example.ServiceSpeaker;
+import org.example.ServiceTyre;
+import org.example.Vehicle;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+@Component
+public class ServiceVehicle {
+    private List<Vehicle> vehicles = new ArrayList<>();
+    public Vehicle findMostExpensiveVehicle(){
+        return vehicles.stream()
+                .sorted(Comparator.comparingDouble(Vehicle::getPrice).reversed())
+                .limit(1)
+                .findFirst()
+                .orElse(null);
+
+    }
+
+    @PostConstruct
+    public void addVehicleToList(){
```

Comment:
1. Variable naming can be better
2. Abstract out in some object creator/generator functions